src/Language/PureScript/TypeChecker.hs

```diff
@@ -193,7 +193,7 @@ checkTypeClassInstance cls i = check where
     TypeLevelString _ _ -> return ()
     TypeConstructor _ ctor -> do
       env <- getEnv
-      when (ctor `M.member` typeSynonyms env) . throwError . errorMessage $ TypeSynonymInstance
+      when (ctor `M.member` typeSynonyms env && not isFunDepDetermined) . throwError . errorMessage $ TypeSynonymInstance
```

Comment:
Is removing this `&& not isFunDepDetermined` the only change necessary to clean up the PR for 0.14?